extensions/amp-a4a/0.1/amp-a4a.js

```diff
@@ -422,15 +457,23 @@ export class AmpA4A extends AMP.BaseElement {
       if (rendered instanceof Error) {
         // If we got as far as getting a URL, then load the ad, but note the
         // error.
-        if (this.adUrl_) {
+        if (this.shouldRenderViaSafeFrame_ == 'safeframe' &&
+            this.creativeBody_) {
+          this.renderViaSafeFrame_();
+        } else if (this.adUrl_) {
           this.renderViaCrossDomainIframe_(true);
         }
         throw rendered;
       }
       if (!rendered) {
         // Was not AMP creative so wrap in cross domain iframe.  layoutCallback
         // has already executed so can do so immediately.
-        this.renderViaCrossDomainIframe_(true);
+        if (this.shouldRenderViaSafeFrame_ == 'safeframe' &&
+            this.creativeBody_) {
+          this.renderViaSafeFrame_();
+        } else {
+          this.renderViaCrossDomainIframe_(true);
+        }
```

Comment:
Can you release this.creativeBody_ after this? I'd consider passing it down as an argument.